java/arcs/core/storage/StorageProxy.kt

```diff
@@ -264,13 +264,14 @@ class StorageProxy<Data : CrdtData, Op : CrdtOperationAtTime, T>(
      * Attempting to perform an operation on a closed [StorageProxy] will result in an exception
      * being thrown.
      */
-    fun close() {
+    suspend fun close() {
         if (stateHolder.value.state == ProxyState.CLOSED) return
+        waitForIdle()
+        store.close()
```

Comment:
Might be even safer if this line went within the `launch` below?

Reply:
I'm running 50k tests right now, it they pass, I'll move it.